api/base/views.py

```diff
@@ -638,12 +639,32 @@ def bulk_get_file_nodes_from_wb_resp(self, files_list):
             )
 
             # mirrors BaseFileNode get_or_create
+            _path = '/' + attrs['path'].lstrip('/')
             try:
-                file_obj = base_class.objects.get(target_object_id=node.id, target_content_type=content_type, _path='/' + attrs['path'].lstrip('/'))
+                file_obj = base_class.objects.get(
+                    target_object_id=node.id,
+                    target_content_type=content_type,
+                    _path=_path,
+                )
             except base_class.DoesNotExist:
                 # create method on BaseFileNode appends provider, bulk_create bypasses this step so it is added here
-                file_obj = base_class(target=node, _path='/' + attrs['path'].lstrip('/'), provider=base_class._provider)
+                file_obj = base_class(target=node, _path=_path, provider=base_class._provider)
                 objs_to_create[base_class].append(file_obj)
+            except MultipleObjectsReturned as e:
+                # Dataverse provides us two sets of files with the same path, so we disambiguate the tiles, this
+                # preserves legacy behavior.
+                if attrs['provider'] == 'dataverse':
+                    dataverse_files = base_class.objects.filter(
+                        target_object_id=node.id,
+                        target_content_type=content_type,
+                        _path=_path,
+                    )
+                    for dataverse_file in dataverse_files:
+                        dataverse_file.update(None, attrs, user=self.request.user, save=False)
+                        file_objs.append(dataverse_file)
+                    continue
```

Comment:
what does this `continue` do? It's the last statement in a block

Reply:
Not quite, still need to skip line 671.